### ISSUE TITLE
[InstanceNorm] Unexpected behaviour with track_running_stats set to True in evaluation mode

### ISSUE THREAD

**Issue by hwangdeyu** [pytorch/pytorch#58843]
## 🐛 Bug

We find the result of the InstanceNorm and batchnorm will get the same result when set **track_running_stats=True** and use **model.eval()**. Since instancenorm 2d is doing normalization to each images whereas batchnorm is doing it to whole minibatch, instancenorm 2d should have more statistics than that of batchnorm. 

## To Reproduce

1. run this code:
```
import torch
import torch.nn as nn

Channel = 2
class TestInstanceNorm(nn.Module):
    def __init__(self):
        super(TestInstanceNorm, self).__init__()
        self.instancenorm = nn.InstanceNorm2d(Channel, affine=True, track_running_stats=True)

    def forward(self, x):
        return self.instancenorm(x)

instance_model = TestInstanceNorm()
instance_model.eval()

class TestBatchNorm(nn.Module):
    def __init__(self):
        super(TestBatchNorm, self).__init__()
        self.batchnorm = nn.BatchNorm2d(Channel, affine=True, track_running_stats=True)

    def forward(self, x):
        return self.batchnorm(x)

batch_model = TestBatchNorm()
batch_model.eval()

dummy_input=torch.randn((3,Channel,1,3))

with torch.no_grad():
    instance_out = instance_model(dummy_input)

with torch.no_grad():
    batch_out = batch_model(dummy_input)

print(instance_out == batch_out)

```

2. get the output:
```
tensor([[[[True, True, True]],

         [[True, True, True]]],


        [[[True, True, True]],

         [[True, True, True]]],


        [[[True, True, True]],

         [[True, True, True]]]], dtype=torch.bool)
```

## Expected behavior

the output of Batchnorm2d and InstanceNorm2d should be different when track_running_stats=True.

## Environment

 - PyTorch Version (e.g., 1.0): 1.9.0a0
 - OS (e.g., Linux): ubuntu 18.04
 - How you installed PyTorch (`conda`, `pip`, source): source
 - Build command you used (if compiling from source): python setuo.py develop
 - Python version: 3.8.8
 - CUDA/cuDNN version: No CUDA

## Additional context
This is another werid point. The internal implement of the instance_norm delegates to batch_norm. If I understand this code [pytorch/aten/src/ATen/native/Normalization.cpp](https://github.com/pytorch/pytorch/blob/5c7dace309cc84cc17629172ce97566285a9da58/aten/src/ATen/native/Normalization.cpp#L552) correctly, the running_mean_ and running_var just repeat it C times and get the shape of new running_mean_ and running_var_ . For example, if the instancenorm input shape is [N, C, W, H], the initial running_mean shape is [C], howevre the computed mean shape will be [N, C] after repeat. So is it reasonable to just repeat the running_mean and running_var and call at:batch_norm to impletment the instance_norm but not calculate it by the value of the input tensor? 

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411

**Comment by zou3519** [comment]
This sounds like a silent correctness issue so I marked it as high-priority but someone with more context behind the normalization layers should take a look at this.

**Comment by jbschlosser** [comment]
I think this behavior is expected, given that no training has been done yet and the modules will be initialized with exactly the same weights (ones and zeros): https://github.com/pytorch/pytorch/blob/2dda8d757192bbfa305eea5df12f0feacb1ffbbf/torch/nn/modules/batchnorm.py#L72-L76

**Comment by H-Huang** [comment]
User error

**Comment by hwangdeyu** [comment]
> I think this behavior is expected, given that no training has been done yet and the modules will be initialized with exactly the same weights (ones and zeros):
> 
> https://github.com/pytorch/pytorch/blob/2dda8d757192bbfa305eea5df12f0feacb1ffbbf/torch/nn/modules/batchnorm.py#L72-L76
> 
> I'd expect the results between `instance_norm` and `batch_norm` to diverge once the `running_mean` / `running_var` values have received training updates.

Thank you for your reply. So you mean track_running_stats == True in evaluation mode would get different result between batchnorm and instancenorm when  the `running_mean` and  `running_var` not equal to ones and zeros. My example is just a special situation right?

**Comment by jfsantos** [comment]
I think you tagged the wrong person ;)

**Comment by hwangdeyu** [comment]
@jbschlosser I tried to run the two normalization twice as below code. it's still get the same true outputs. Do you think it also expected results?

```
import torch
import torch.nn as nn
 
Channel = 2
torch.manual_seed(7)
instance_model = nn.InstanceNorm2d(Channel, affine=True, track_running_stats=True)
instance_model.eval()

torch.manual_seed(7)
batch_model = nn.BatchNorm2d(Channel, affine=True, track_running_stats=True)
batch_model.eval()

dummy_input=torch.randn((3,Channel,1,3))
# print("dummy_input:", dummy_input)
 
with torch.no_grad():
    instance_out_1 = instance_model(dummy_input)
    instance_out_2 = instance_model(dummy_input)
 
with torch.no_grad():
    batch_out_1 = batch_model(dummy_input)
    batch_out_2 = batch_model(dummy_input)
    
print("result_1:", (instance_out_1 == batch_out_1).to(torch.bool))
print("result_2:", (instance_out_2 == batch_out_2).to(torch.bool))
```

output result:
```
result_1: tensor([[[[True, True, True]],
         [[True, True, True]]],
        [[[True, True, True]],
         [[True, True, True]]],
        [[[True, True, True]],
         [[True, True, True]]]])
result_2: tensor([[[[True, True, True]],
         [[True, True, True]]],
        [[[True, True, True]],
         [[True, True, True]]],
        [[[True, True, True]],
         [[True, True, True]]]])
```

**Comment by hwangdeyu** [comment]
> > I think this behavior is expected, given that no training has been done yet and the modules will be initialized with exactly the same weights (ones and zeros):
> > https://github.com/pytorch/pytorch/blob/2dda8d757192bbfa305eea5df12f0feacb1ffbbf/torch/nn/modules/batchnorm.py#L72-L76
> > 
> > I'd expect the results between `instance_norm` and `batch_norm` to diverge once the `running_mean` / `running_var` values have received training updates.
> 
> @jfsantos Thank you for your reply. So you mean track_running_stats == True in evaluation mode would get different result between batchnorm and instancenorm when the `running_mean` and `running_var` not equal to ones and zeros. My example is just a special situation right?

@jbschlosser Could you please help me to take a look this two replies above? thanks!

**Comment by jbschlosser** [comment]
> @jbschlosser Could you please help me to take a look this two replies above? I am waitting for your reply, thanks!

Hey @hwangdeyu, to clarify my previous response, you have to do some training on the affine layer parameters within `InstanceNorm2d` / ` BatchNorm2d` for the results to be different. In your example, you never train these parameters, and so they remain at the same default value.

Adding some "training" to your code above demonstrates the difference between the two. Note that even though the two models are trained with the same inputs and the same optimizer settings, they produce different final results since they are two different norms:

```python
import torch
import torch.nn as nn

Channel = 2
torch.manual_seed(7)
instance_model = nn.InstanceNorm2d(Channel, affine=True, track_running_stats=True)

torch.manual_seed(7)
batch_model = nn.BatchNorm2d(Channel, affine=True, track_running_stats=True)

# Do some "training".
instance_optimizer = torch.optim.Adam(instance_model.parameters())
batch_optimizer = torch.optim.Adam(batch_model.parameters())
for _ in range(10):
    dummy_input = torch.randn((3,Channel,1,3))
    instance_optimizer.zero_grad()
    batch_optimizer.zero_grad()
    instance_model(dummy_input).sum().backward()
    batch_model(dummy_input).sum().backward()
    instance_optimizer.step()
    batch_optimizer.step()

instance_model.eval()
batch_model.eval()

with torch.no_grad():
    instance_out_1 = instance_model(dummy_input)
    instance_out_2 = instance_model(dummy_input)

with torch.no_grad():
    batch_out_1 = batch_model(dummy_input)
    batch_out_2 = batch_model(dummy_input)

print("result_1:", (instance_out_1 == batch_out_1).to(torch.bool))
print("result_2:", (instance_out_2 == batch_out_2).to(torch.bool))
```

```
result_1: tensor([[[[False, False, False]],

         [[False, False, False]]],


        [[[False, False, False]],

         [[False, False, False]]],


        [[[False, False, False]],

         [[False, False, False]]]])
result_2: tensor([[[[False, False, False]],

         [[False, False, False]]],


        [[[False, False, False]],

         [[False, False, False]]],


        [[[False, False, False]],

         [[False, False, False]]]])
```
